sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -357,6 +375,15 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
     val canUseV2 = lookupV2Provider().isDefined
     val sessionCatalog = session.sessionState.analyzer.sessionCatalog
 
+    var parition = Map[String, Option[String]]()
+    if (partionInfo.isDefined) {
+      val res = partionInfo.get.foreach(partion => {
```

Comment:
Personally, I'd prefer us to expose the Map rather than doing string parsing. Unless is this matching an API in another system which is build like this?
This is more of my personal preference, but it's easy for us to make mistakes parsing strings so when I don't have to it's nice to avoid.